spec/function_object_spec.rb

```diff
@@ -27,11 +27,39 @@ def call
     end
   end
 
+  let(:simple) do
+    def_function do
+      args :a, :b
+
+      def call
+        a + b
+      end
+    end
+  end
+
+  let(:duplicating) do
+    def_function do
+      args :a do
+        arg :a
+      end
+    end
+  end
+
   describe '.call' do
     it 'does something useful' do
       expect(plus.(1,2)).to eq 3
     end
 
+    it 'even with simple argument list' do
+      expect(simple.(1,2)).to eq 3
+    end
+
+    it 'duplicating arguments raises error' do
+        expect {
+          duplicating.(1)
```

Comment:
I think here `expect { duplicating }` would be enough. And it's even more readable since we describe a definition error, not a call error.